database/votechange.go

```diff
@@ -27,13 +27,14 @@ type VoteChangeRecord struct {
 // one which is currently stored. Records are stored using a serially increasing
 // integer as the key.
 func (vdb *VspDatabase) SaveVoteChange(ticketHash string, record VoteChangeRecord) error {
+	// Marshal record first. This could prevent a DB transaction being started
+	// and rolled back unnecessarily.
+	recordBytes, err := json.Marshal(record)
```

Comment:
I agree with the marshal being outside of the db tx here, potentially prevents an rollback as intended.